lib/security/plugs.ex

```diff
@@ -6,11 +6,46 @@ defmodule PhoenixApiToolkit.Security.Plugs do
   """
   alias Plug.Conn
   import Plug.Conn
-  alias PhoenixApiToolkit.Security.{MissingContentTypeError, Oauth2TokenVerificationError}
+
+  alias PhoenixApiToolkit.Security.{
+    MissingContentTypeError,
+    Oauth2TokenVerificationError,
+    AjaxCSRFError
+  }
+
   require Logger
 
-  # DELETE should not have a request body
-  @unsafe_methods ~w(PUT POST PATCH) |> MapSet.new()
+  @doc """
```

Comment:
Ik begrijp nog steeds niet hoe dit veiliger is, als ik een verbinding wil maken kan ik deze toch gewoon zetten?

Reply:
This repo is public ;) I got it from https://cheatsheetseries.owasp.org/cheatsheets/REST_Security_Cheat_Sheet.html#validate-request-content-types which recommends rejecting requests without an explicit content-type set.

Reply:
The idea is that only javascript is allowed to set custom headers (which "x-csrf-token" is) by the same-origin policy of browsers. A forged website that posts fake data on behalf of a user cannot set this header because it cannot be hosted on the correct origin.

Reply:
check, I keep looking at this from the server perspective but from a browser perspective that makes a lot more sense

Reply:
The API server just has to make sure it doesn't do something colossally stupid like putting allow-all-origins CORS headers ;)